src/github/logic.py

```diff
@@ -307,11 +304,36 @@ def maybe_automerge_pull_request_and_rerun_stale_checks(
     return False
 
 
+def maybe_rerun_stale_checks_on_approved_pull_request(
+    pull_request: PullRequest,
+) -> bool:
+    if (
+        SGTM_FEATURE__ALLOW_CHECK_RERUN_ON_APPROVAL
+        and _pull_request_is_open(pull_request)
+        and pull_request.is_approved()
+    ):
+        logger.info(
+            f"PR-{pull_request.id()} is open and approved, maybe rerun stale checks"
+        )
+        return _maybe_rerun_stale_checks(pull_request)
+    logger.info(
+        f"SGTM_FEATURE__ALLOW_CHECK_RERUN_ON_APPROVAL is {SGTM_FEATURE__ALLOW_CHECK_RERUN_ON_APPROVAL}"
+    )
+    logger.info(
+        f"{pull_request.id()} is {'' if _pull_request_is_open(pull_request) else 'not '}open and {'' if pull_request.is_approved() else 'not '}approved"
+    )
+    return False
+
+
 # ----------------------------------------------------------------------------------
 # Automerge helpers
 # ----------------------------------------------------------------------------------
 
 
+def _pull_request_is_open(pull_request: PullRequest) -> bool:
```

Comment:
not a requested change: What's the decision behind adding a helper to determine pull request open state vs. adding a class method to `PullRequest` to do it and to query it as a field? i.e. `pull_request.open()` or `pull_request.is_open()`

Reply:
Mainly because the `PullRequest` class methods get values directly from the API, which is not what we're doing here